src/App.jsx

```diff
@@ -1,24 +1,70 @@
-import logo from "./assets/technigo-logo.svg";
-// Import the custom hook useQRCodeGenerator
-import { QrExample } from "./components/QrExample";
+import { useQRCodeGenerator } from "./hooks/useQRCodeGenerator";
+import { MyLottieAnimation } from "./components/LottieQR";
 
-// Define the App component
+// App Component
 export const App = () => {
-  // Destructure variables, properties and methods from the useQRCodeGenerator hook that you imported above here :)
+  // Destructure variables, properties, and methods from the useQRCodeGenerator hook
+  const {
+    url,
+    setUrl,
+    qr,
+    showInput,
+    errorMessage,
+    generateQRCode,
+    downloadQRCode,
+    repeatAction,
+    qrGenerated,
+  } = useQRCodeGenerator();
 
-  // Return the JSX to render the component
+  // Return JSX to render the component
   return (
-    <div className="">
-      {/* Render the title */}
-      <img className="logo" src={logo} alt="" />
-      <h1>Technigo QR Code Generator</h1>
-      <p>Start Here</p>
-      <QrExample />
-
-      {/* Conditionally render based on wether the user is inputting an URL to generate a QR Code or the user wnats to downaload the generated QR Code from the url input */}
-      {/* {yourReactiveVariableThatTogglesTheDownloadQrCcodeOrInputField ? () : ()} */}
+    <div>
+      <nav className="navbar">
+        <h1>QR Code Generator</h1>
+      </nav>
+      <div className="pageWrapper">
+        {qrGenerated ? null : <MyLottieAnimation />} {/* Conditionally render MyLottieAnimation */}
+        {/* Conditionally render based on whether the user is inputting a URL to generate a QR Code or wants to download the generated QR Code */}
+        {showInput ? (
+          <>
+            <input
+              type="text"
+              value={url}
+              onChange={(e) => {
+                setUrl(e.target.value);
+                if (errorMessage) setErrorMessage('');
+              }}
+              onKeyDown={(e) => {
+                if (e.key === 'Enter') {
```

Comment:
I love this! I always use enter instead of clicking the button, so I really appreciate it.